RecoBTag/FeatureTools/plugins/UnifiedParticleTransformerAK4TagInfoProducer.cc

```diff
@@ -0,0 +1,522 @@
+      vtx_token_(consumes<VertexCollection>(iConfig.getParameter<edm::InputTag>("vertices"))),
+      lt_token_(consumes<pat::PackedCandidateCollection>(iConfig.getParameter<edm::InputTag>("losttracks"))),
+      sv_token_(consumes<SVCollection>(iConfig.getParameter<edm::InputTag>("secondary_vertices"))),
+      candidateToken_(consumes<edm::View<reco::Candidate>>(iConfig.getParameter<edm::InputTag>("candidates"))),
+      track_builder_token_(
+          esConsumes<TransientTrackBuilder, TransientTrackRecord>(edm::ESInputTag("", "TransientTrackBuilder"))),
+      use_puppi_value_map_(false),
+      use_pvasq_value_map_(false),
+      use_unsubjet_map_(false),
+      fallback_puppi_weight_(iConfig.getParameter<bool>("fallback_puppi_weight")),
+      fallback_vertex_association_(iConfig.getParameter<bool>("fallback_vertex_association")),
+      is_weighted_jet_(iConfig.getParameter<bool>("is_weighted_jet")),
+      min_jet_pt_(iConfig.getParameter<double>("min_jet_pt")),
+      max_jet_eta_(iConfig.getParameter<double>("max_jet_eta")) {
+  produces<UnifiedParticleTransformerAK4TagInfoCollection>();
+
+  const auto& puppi_value_map_tag = iConfig.getParameter<edm::InputTag>("puppi_value_map");
+  if (!puppi_value_map_tag.label().empty()) {
+    puppi_value_map_token_ = consumes<edm::ValueMap<float>>(puppi_value_map_tag);
+    use_puppi_value_map_ = true;
+  } else if (is_weighted_jet_) {
+    throw edm::Exception(edm::errors::Configuration,
+                         "puppi_value_map is not set but jet is weighted. Must set puppi_value_map.");
+  }
+
+  const auto& pvas_tag = iConfig.getParameter<edm::InputTag>("vertex_associator");
+  if (!pvas_tag.label().empty()) {
+    pvasq_value_map_token_ = consumes<edm::ValueMap<int>>(pvas_tag);
+    pvas_token_ = consumes<edm::Association<VertexCollection>>(pvas_tag);
+    use_pvasq_value_map_ = true;
+  }
+
+  const auto& unsubjet_map_tag = iConfig.getUntrackedParameter<edm::InputTag>("unsubjet_map", {});
+  if (!unsubjet_map_tag.label().empty()) {
+    unsubjet_map_token_ = consumes<JetMatchMap>(unsubjet_map_tag);
+    use_unsubjet_map_ = true;
+  }
+}
+
+void UnifiedParticleTransformerAK4TagInfoProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // pfUnifiedParticleTransformerAK4TagInfos
+  edm::ParameterSetDescription desc;
+  desc.add<double>("jet_radius", 0.4);
+  desc.add<double>("min_candidate_pt", 0.10);
+  desc.add<bool>("flip", false);
+  desc.add<edm::InputTag>("vertices", edm::InputTag("offlinePrimaryVertices"));
+  desc.add<edm::InputTag>("losttracks", edm::InputTag("lostTracks"));
+  desc.add<edm::InputTag>("puppi_value_map", edm::InputTag("puppi"));
+  desc.add<edm::InputTag>("secondary_vertices", edm::InputTag("inclusiveCandidateSecondaryVertices"));
+  desc.add<edm::InputTag>("jets", edm::InputTag("ak4PFJetsCHS"));
+  desc.addUntracked<edm::InputTag>("unsubjet_map", {});
```

Comment:
```suggestion
  desc.add<edm::InputTag>("unsubjet_map", {});
```
I think it should be "tracked", see https://github.com/cms-sw/cmssw/pull/44591.

Reply:
Thanks I will adjust it :+1: